tests/functional/NotificationTargetKnowbaseItem.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+use Config;
+use DbTestCase;
+use Group;
+use Notification;
+use NotificationTarget;
+use QueuedNotification;
+
+/* Test for inc/notificationtargetticket.class.php */
+
+class NotificationTargetKnowbaseItem extends DbTestCase
+{
+    public function testgetDataForNotifKnowbaseItem()
+    {
+        $this->login();
+
+        Config::setConfigurationValues('core', ['use_notifications' => 1]);
+        Config::setConfigurationValues('core', ['notifications_mailing' => 1]);
+
+        //set notification by mail to active
+        $notif = new Notification();
+        $knowbasenotifs = $notif->find(
+            [
+                'itemtype' => 'KnowbaseItem'
+            ]
+        );
+        // test activate notification
+        foreach ($knowbasenotifs as $kbnotif) {
+            $datawant = $kbnotif;
+            $datawant['is_active'] = 1;
+            $notif->update($datawant);
+            $kbnotif = $notif->getById($kbnotif['id'])->fields;
+            $kbnotif['date_mod'] = null;
+            $this->array($kbnotif)->isIdenticalTo($datawant);
```

Comment:
Following code should be enough to ensure that notifications are active.
```suggestion
            $this->boolean($notif->update(['id' => $kbnotif['id'], 'is_active' => 1]))->isTrue();
```